app/models/concerns/has_deprecated_columns.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module HasDeprecatedColumns
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def deprecate_column(column_name, drop_from:, raise_error: false, message: nil)
+      if Gem::Version.new(Discourse::VERSION::STRING) >= Gem::Version.new(drop_from)
+        self.ignored_columns = self.ignored_columns.dup << column_name.to_s
```

Comment:
This is frozen by Rails when assigned, so we need to duplicate the original array first to support mutation.